saleor/graphql/product/types/products.py

```diff
@@ -727,15 +732,42 @@ def resolve_images(root: ChannelContext[models.Product], info, **_kwargs):
 
     @staticmethod
     def resolve_variants(root: ChannelContext[models.Product], info, **_kwargs):
+        requestor = get_user_or_app_from_context(info.context)
+        is_staff = requestor_is_staff_member_or_app(requestor)
+
+        def return_available_variants_in_channel(variants):
+            if is_staff:
+                return variants
+
+            variants_id = (variant.id for variant in variants)
+            variant_channel_listings = (
```

Comment:
`variant_channel_listings` - This dataloader doesn't return ProductVariantChannelListing